app/assets/javascripts/school_administrator_dashboard/dashboard_components/tardies_dashboard/TardiesDataLoader.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import SchoolwideTardies from './SchoolwideTardies';
+import {apiFetchJson} from '../../../helpers/apiFetchJson';
+import measurePageLoad from '../../../helpers/measurePageLoad';
+
+class TardiesDataLoader extends React.Component {
+  //This class just grabs tardy data from the server endpoint before loading the rest of the tardies dashboard
+  constructor(props) {
+    super(props);
+    this.state = {
+      dashboardStudents: this.props.dashboardStudents,
+      isLoading: false
+    };
+  }
+
+  componentWillMount() {
+    //if the parent component already has the dashboard data, we don't need to duplicate it
+    if(this.props.dashboardStudents.length) return;
+    const url = `/schools/${this.props.schoolId}/tardies_dashboard_data`;
+    this.setState({isLoading: true});
+
+    apiFetchJson(url)
```

Comment:
There's a `<GenericLoader>` component if you want ([example](https://github.com/studentinsights/studentinsights/blob/ccba754239820bac5371275b25f56be0ed0537b7/app/assets/javascripts/school_courses/SchoolCoursesPage.js#L31)), not sure if it's actually useful or an improvement over the explicit code here, probably the only thing you get is a generic error message (not handled here) and a default loading message.

Reply:
Neat, I didn't see that!